src/OpenConext/UserLifecycle/Application/Service/LastLoginService.php

```
@@ -0,0 +1,81 @@
+ * limitations under the License.
+ */
+
+namespace OpenConext\UserLifecycle\Application\Service;
+
+use InvalidArgumentException;
+use OpenConext\UserLifecycle\Domain\Client\DeprovisionClientCollectionInterface;
+use OpenConext\UserLifecycle\Domain\Query\LastLoginByUserIdQuery;
+use OpenConext\UserLifecycle\Domain\QueryHandler\LastLoginByUserIdQueryHandlerInterface;
+use OpenConext\UserLifecycle\Domain\Service\LastLoginServiceInterface;
+use OpenConext\UserLifecycle\Domain\ValueObject\CollabPersonId;
+use Psr\Log\LoggerInterface;
+use Webmozart\Assert\Assert;
+
+class LastLoginService implements LastLoginServiceInterface
+{
+    /**
+     * @var LastLoginByUserIdQueryHandlerInterface
+     */
+    private $lastLoginByUserIdQueryHandler;
+
+    /**
+     * @var DeprovisionClientCollectionInterface
+     */
+    private $deprovisionClientCollection;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(
+        LastLoginByUserIdQueryHandlerInterface $queryHandler,
+        DeprovisionClientCollectionInterface $deprovisionClientCollection,
+        LoggerInterface $logger
+    ) {
+        $this->lastLoginByUserIdQueryHandler = $queryHandler;
+        $this->deprovisionClientCollection = $deprovisionClientCollection;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param string $personId
+     * @throws InvalidArgumentException
+     */
+    public function readInformationFor($personId)
+    {
+        $this->logger->debug('Received a request for user information');
+
+        Assert::stringNotEmpty($personId, 'Please pass a non empty collabPersonId');
+
+        $collabPersonId = new CollabPersonId($personId);
+        $query = new LastLoginByUserIdQuery($collabPersonId);
+
+        $this->logger->debug('Finding the last login entry for the user.');
+        $lastLogin = $this->lastLoginByUserIdQueryHandler->handle($query);
+
+        Assert::notNull(
```

Seems to me that in the case you want to throw an exception here you want to have a more specific NoInformationFoundException ? But, not sure maybe its OK to just return nothing/null?